File: Code Samples/CMSC 21/1. Loops or Iterations/3. for loop/for-loop.c
#include <stdio.h>

/* This code is a for-loop version of the original version from item #2. */

// Variable declarations
int num_variable = 0;
const int exit_value = 10;

// For-loop version
int main() {

    // start of for-loop
    for (int current_number=0; current_number<10; current_number++) {
        num_variable += 1;            // Update the number
        printf("%d\n", num_variable); // Print the number
    }

    // End of code
    return 0;
}



/* further explanation below for your reference


#====== for-loop version (1) ======#        #========= linear version (1) =========#

int num_variable = 0;                       int num_variable = 0

for (int i=0; i<1; i++) {                   // iteration 1
    num_variable++;                         num_variable ++;                             
    printf("%d\n", num_variable);           printf("%d\n", num_variable);
}

#==================================#        #======================================#





#====== for-loop version (3) ======#        #========= linear version (3) =========#

int num_variable = 0;                       int num_variable = 0

for (int i=0; i<3; i++) {                   // iteration 1
    num_variable++;                         num_variable ++;                             
    printf("%d\n", num_variable);           printf("%d\n", num_variable);
}

                                            // iteration 2
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


                                            // iteration 3
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


#==================================#        #======================================#





#====== for-loop version (10) =====#        #========= linear version (10) ========#

int num_variable = 0;                       int num_variable = 0

for (int i=0; i<10; i++) {                  // iteration 1
    num_variable++;                         num_variable ++;                             
    printf("%d\n", num_variable);           printf("%d\n", num_variable);
}

                                            // iteration 2
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


                                            // iteration 3
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


                                            // iteration 4
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


                                            // iteration 5
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


                                            // iteration 6
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


                                            // iteration 7
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


                                            // iteration 8
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


                                            // iteration 9
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);


                                            // iteration 10
                                            num_variable ++;                             
                                            printf("%d\n", num_variable);

#==================================#        #======================================#


'''

*/

File: Code Samples/CMSC 12/1. Loops or Iterations/3. loop versions/3.1. for-loop/for-loop.py
''' This code is a for-loop version of the original version from item 3.0. '''

# variable declarations
num_variable = 0

# STOP if already in this value!
exit_value = 10


# for-loop version 
for current_number in range(10):    # <- take note! 'current_number' is a different variable now
    num_variable += 1               # <- update the number
    print(num_variable)             # <- print the number



# further explanation below for your reference

'''
#====== for-loop version (1) ======#        #========= linear version (1) =========#

num_variable = 0                            num_variable = 0

for current_number in range(1):             # iteration 1
    num_variable += 1                       num_variable += 1
    print(num_variable)                     print(num_variable)     <- prints what?

#==================================#        #======================================#





#====== for-loop version (3) ======#        #========= linear version (3) =========#

num_variable = 0                            num_variable = 0

for current_number in range(3):             # iteration 1
    num_variable += 1                       num_variable += 1
    print(num_variable)                     print(num_variable)     <- prints what?
    
                                            # iteration 2
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # iteration 3
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
#==================================#        #======================================#





#====== for-loop version (10) =====#        #========= linear version (10) ========#

num_variable = 0                            num_variable = 0

for current_number in range(10):             # iteration 1
    num_variable += 1                       num_variable += 1
    print(num_variable)                     print(num_variable)     <- prints what?
    
                                            # iteration 2
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # iteration 3
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
        
                                            # iteration 4
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # iteration 5
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
        
                                            # iteration 6
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # iteration 7
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
        
                                            # iteration 8
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # iteration 9
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
        
                                            # iteration 10
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
#==================================#        #======================================#


'''